### DISCUSSION TITLE
USB hub not working with Pi Pico

### DISCUSSION THREAD

**Discussion by jeroentaverne** [hathach/tinyusb#1248] [Q&A]
I got issues with a USB hub.
I have connected a USB keyboard and with the USB host examples this works fine. When I connect a random USB hub in between it always fails. No events can be seen from the keyboard. I tried multiple hubs and keyboards. See attached logs and tusb_config.h. Hope you could help me out. Thanks!
[hub+keyboard.txt](https://github.com/hathach/tinyusb/files/7735664/hub%2Bkeyboard.txt)
[onlyhub.txt](https://github.com/hathach/tinyusb/files/7735665/onlyhub.txt)
[onlykeyboard.txt](https://github.com/hathach/tinyusb/files/7735666/onlykeyboard.txt)
[tusb_config.txt](https://github.com/hathach/tinyusb/files/7735669/tusb_config.txt)
.

**Comment by rppicomidi** [comment]
I filed issue #1250. I am seeing issues with a hub and keyboard too.

**Reply by jeroentaverne** [reply]
Thanks for reporting more info  The logfiles seem to be very similar.
